vendor/plugins/edu.harvard.dce.paella.iFrameApiAppraiser/iFrameApiAppraiser.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * DCE IFrameApiHelper
+ * IFrame embed API plugin used in conjunction with iFrameEmbedApi.js
+ * that is imported into parent window.
+ * @author HUDCE
+ */
+paella.addPlugin(function () {
+  return class IFrameApiAppraiser extends paella.EventDrivenPlugin {
+    /**
+    * @constructor
+    * @augments paella.EventDrivenPlugin
+    */
+    constructor () {
+      super();
+      // This bind must happen early to catch the videoReady event
+      paella.events.bind(
+        paella.events.videoReady,
+        () => {
+          this.sendMessageToEmbedApi(paella.events.videoReady);
+        }
+      );
+    }
+
+    /**
+     * @inheritdoc
+     * Get the name of the Paella plugin
+     * @return the name of this plugin
+     */
+    getName() {
+      return 'edu.harvard.dce.paella.iFrameApiAppraiser';
+    }
+
+    /**
+      * @inheritdoc
+      * Get the name of the Paella plugin
+      * @param {requestCallback} onSuccess - The callback that handles response
+      */
+    checkEnabled (onSuccess) {
+      // This iFrame prefix is used by the iFrameEmbedApi
+      // When constructing the embeded player iFrame
+      const iFramePrefix = 'DCE-iframe-API';
+      // Enable when player is embedded, and embed iFrame has target prefix
+      onSuccess(
+        (window.self !== window.top)
+        && (window.name && window.name.startsWith(iFramePrefix))
+      );
+    }
+
+    /**
+      * @inheritdoc
+      * Get the events that this Paella event driven plugin uses
+      * @return an array of events
+      */
+    getEvents() {
+      return [
+        paella.events.play,
+        paella.events.pause,
+        paella.events.loadComplete,
+        // paella.events.videoReady is bound early above
+        paella.events.seeked,
+        paella.events.timeupdate,
+        paella.events.error,
+        paella.events.ended
+      ];
+    }
+
+    /**
+      * @inheritdoc
+      * The handle this Paella event driven plugin is
+      * called when an event fires.
+      * @param {string} eventType - the event that was fired
+      * @param {object} [params] - a set of params associated to the event
+      */
+    onEvent(eventType, params) {
+      switch (eventType) {
+        case paella.events.timeupdate:
+          this.sendMessageToEmbedApi(eventType, params.currentTime);
+          break;
+        case paella.events.play:
+        case paella.events.pause:
+        case paella.events.seeked:
+        case paella.events.error:
+        case paella.events.videoReady:
+        case paella.events.loadComplete:
```

Comment:
Does this really work as intended without `break;` statements?

Perhaps switch this to a set of `ifs`?

Reply:
The switch seemed cleaner than 3 if-thens.
The list of cases are just the fall-through of cases that all fall to the this.sendMessageToEmbedApi(eventType);
Anything not in that list is not supported to send up to the container.

Reply:
No I mean that this logic might not work as intended without `break;` statements.

For example, this switch will call this line: `paella.log.debug('IFrameApiAppraiser: Unsupported event ', eventType);` for anything that's not `paella.events.timeupdate` or `paella.events.ended`

Reply:
Line 87 has the break for all the cases above it that don't have a break. They all fall through to the paella.events.ended.  Then hit that break at line 87. That's the beauty of a switch.

Reply:
Switch is cool to do stuff like :
case: is formal dinner
   dress nice
case: is dinner
  eat a light lunch to save room for dinner
  break;
.... 
So the formal dinner gets both dress nice and eat a light lunch.

Reply:
Is it intended behavior for `paella.events.videoReady` to print "Unsupported event"?

Reply:
Here's a case that shows 2 case catches getting the same action https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
(just an example)

Reply:
> 



> Is it intended behavior for `paella.events.videoReady` to print "Unsupported event"?

If it was the `this.sendMessageToEmbedApi(eventType);` would be moved below the ` this.sendMessageToEmbedApi(eventType);` which is what it's hitting now. Or remove from the CASE so it automatically goes to the default catch.

Reply:
If you want `paella.events.play`, `paella.events.pause`, `paella.events.seeked`, `paella.events.error`, `paella.events.videoReady`, and `paella.events.loadComplete` to call `paella.log.debug` and log "Unsupported event" then we're all good and I understand.

Reply:
The break on 87 prevents `paella.events.videoReady` from getting to the debug line.

Reply:
Oh wait. I'm not reading this properly. >.<

Reply:
Okay I get it. Wow sorry that took me so long :P
I still don't like it haha >.< but I understand it now. You can resolve this convo.

Reply:
If I wrote this, I'd probably use something like:

```
// Make sure the event is supported
if (!(eventType in Object.values(paella.events))) {
  paella.log.debug('IFrameApiAppraiser: Unsupported event ', eventType);
}

// Send the notification
if (eventType === paella.events.timeupdate) {
  this.sendMessageToEmbedApi(eventType, params.currentTime);
} else {
  this.sendMessageToEmbedApi(eventType);
}
```

but that's just me.

Again, I get this now. Sorry!

Reply:
I get that. But, there are a lot of events in paella events that are not useful or possibly even outdated and unused. Enumerating the events to pass to the wrapper  gives this plugin control to consolidate the events and protect if the naming changes on the event inside the player. 
There are pros & cons. But as an API, it seems safer to be explicit about what is being be passed to the wrapper.